File: sol_study/other_test/access_noname_table.cpp
//
//  access_noname_table.cpp
//  sol_study
//  测试 获取lua没有名称的表中的元素
//  Created by lifushan on 2018/3/20.
//  Copyright © 2018年 lifs. All rights reserved.
//

#include <stdio.h>

#define SOL_CHECK_ARGUMENTS 1
#include "sol.hpp"
#include "assert.hpp"

int access_noname_table()
{
    sol::state lua;
    lua.open_libraries();
    
    lua.script_file("./other_test/access_noname_table.lua");
    
    printf("\n------------------------ C get no name table test st -------------------------\n\n");
    
    sol::object oo = lua["enemy"];
    sol::table ob = lua["enemy"];
    for (int i = 1; i < 3; ++i) {
        oo = ob[i]["name"];
        std::string name = ob[i]["name"];
        int health = ob[i]["health"];
        printf("--- index[%d], type[%d], name[%s], health[%d] ----\n", i, oo.get_type(), name.c_str(), health);
    }
//    sol::type::function;
    
    printf("\n------------------------ C get no name table test st -------------------------\n\n");
    
    return 0;
}

File: sol_study/other_test/access_noname_table.lua
--
-- http://lua-users.org/wiki/TablesTutorial
-- 
print("\n------------------------ Lua table no name access test en -------------------------\n\n");

enemy = {
    aa = "cc",

    {
        name = "ogre",
        health = 30,
        skill = 10,
        sounds = { "ogresond1", "ogresound2"},
    },
    {
        name = "orc",
        health = 10,
        skill = 20,
        sounds = { "orcsond1", "orcsound2"},
    }
    -- and so on
}

print("index 1 name is " .. enemy[1].name);
print("index 2 name is " .. enemy[2].name);

level1 =
{
    music = "music1",
    items =
    {
        A = 1,
        B = 2,
        {
            c = "C",
        }
    }
}

print("level1 table's items table index 1's c value is " .. level1.items[1].c);

print("\n------------------------ Lua table no name access test en -------------------------\n\n");
